nixos/modules/services/hardware/auto-epp.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+
+ cfg = config.services.auto-epp;
+ package = pkgs.auto-epp;
+
+in {
+ options = {
+
+    services.auto-epp = {
+      enable = mkEnableOption (lib.mdDoc "auto-epp for amd active pstate");
+      batState= mkOption {
+        type = types.str;
+        default = "power";
+        description = lib.mdDoc "energy_performance_preference when on battery";
+      };
+
+     acState= mkOption {
+        type = types.str;
+        default = "balance_performance";
+        description = lib.mdDoc "energy_performance_preference when plugged in";
+      };
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    boot.kernelParams = [
+      "amd_pstate=active"
+    ];
+
+    environment.etc."auto-epp.conf".text = ''
+     [Settings]
+     epp_state_for_AC=${cfg.acState}
+     epp_state_for_BAT=${cfg.batState}
+    '';
+
+    systemd.packages = [ package ];
+
+    systemd.services.auto-epp = {
+      enable = true;
+      after = [ "network.target" "network-online.target"  ];
+      description = "auto-epp - Automatic EPP Changer for amd-pstate-epp";
+      serviceConfig = {
+        Type = "simple";
+        User = "root";
```

Comment:
Does this service need to run as root? Maybe we could make it `DynamicUser` and give it the permissions it needs through `CapabilityBoundingSet`?

Reply:
I'll need to research on this as i have no idea on how those work.

Reply:
After some check, it requires to run as root, logs from journalctl:
```
Oct 17 12:02:01 yoga7 auto-epp[53340]: auto-epp must be run with root privileges.
```